core/src/main/java/com/netflix/conductor/service/WorkflowServiceImpl.java

```diff
@@ -131,14 +131,6 @@ public String startWorkflow(
             WorkflowDef workflowDef) {
 
         if (workflowDef == null) {
-            workflowDef = metadataService.getWorkflowDef(name, version);
```

Comment:
Why was this removed? Workflow definition needs to be pre-registered if not supplied in the start workflow request.

Reply:
The check is redundant. In the call stack, `WorkflowExecutor.startWorkflow` calls [MetadataMapperService.lookupForWorkflowDefinition](https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java#L331-L332) which throws the same [ApplicationException](https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/metadata/MetadataMapperService.java#L66-L77).